app/controllers/org_admin/questions_controller.rb

```diff
@@ -79,15 +73,39 @@ def destroy
         flash[:alert] = _('Unable to create a new version of this template.')
       end
       redirect_to edit_org_admin_template_phase_path({
-        template_id: question.section.phase.template.id, 
-        id: question.section.phase.id, 
-        section: question.section.id
+        template_id: section.phase.template.id, 
+        id: section.phase.id, 
+        section: section.id
       })
     end
 
     private
       def question_params
         params.require(:question).permit(:number, :text, :question_format_id, :option_comment_display, :default_value, question_options_attributes: [:id, :number, :text, :is_default, :_destroy], annotations_attributes: [:id, :text, :org_id, :org, :type], theme_ids: [])
       end
+      
+      # When a template gets versioned by changes to one of its questions we need to loop 
+      # through the incoming params and ensure that the annotations and question_options
+      # get attached to the new question
+      def transfer_associations(question)
+        attrs = question_params
+        if attrs[:annotations_attributes].present?
+          attrs[:annotations_attributes].each_key do |key|
+            old_annotation = question.annotations.select do |a| 
```

Comment:
Use find instead of select since it will return the annotation object or nil instead of an array with all the matches. It also works as a search instead of iterating all the annotation for every key at annotations_attributes....

Reply:
switiched it over but it started failing. Going to leave as-is for now. We can adjust in a separate PR afterward since I'd like to get this out for UAT